From f90e2ed3002faaa434111eb45e2ebaf504834b1e Mon Sep 17 00:00:00 2001
From: Leorize <leorize+oss@disroot.org>
Date: Fri, 11 Jul 2025 14:51:55 -0500
Subject: [PATCH] oraclecloud: do not wrap errors from FetchToBuffer

Ignition internal machinery relies on being able to match against an
error directly, in this case, `ErrNeedNet`. By wrapping errors from
FetchToBuffer, ignition would not be able to detect the lack of
networking during `fetch-offline` phase and fail to run.

Signed-off-by: Leorize <leorize+oss@disroot.org>
Signed-off-by: Tiago Bueno <tiago.bueno@gmail.com>
---
 docs/release-notes.md                         | 1 +
 internal/providers/oraclecloud/oraclecloud.go | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/docs/release-notes.md b/docs/release-notes.md
index da586fc8..bdd8900f 100644
--- a/docs/release-notes.md
+++ b/docs/release-notes.md
@@ -14,6 +14,7 @@ nav_order: 9
 
 ### Bug fixes
 
+- Fix fetch-offline for Oracle Cloud Infrastructure
 
 ## Ignition 2.22.0 (2025-07-08)
 Starting with this release, ignition-validate binaries are signed with the
diff --git a/internal/providers/oraclecloud/oraclecloud.go b/internal/providers/oraclecloud/oraclecloud.go
index 33db95c3..fb06caf5 100644
--- a/internal/providers/oraclecloud/oraclecloud.go
+++ b/internal/providers/oraclecloud/oraclecloud.go
@@ -53,7 +53,8 @@ func fetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 		RetryCodes: []int{http.StatusTooManyRequests},
 	})
 	if err != nil && err != resource.ErrNotFound {
-		return types.Config{}, report.Report{}, fmt.Errorf("fetching to buffer: %w", err)
+		// Do not wrap these errors, ignition uses direct comparsion to distinguish them.
+		return types.Config{}, report.Report{}, err
 	}
 
 	userdata := make([]byte, base64.StdEncoding.DecodedLen(len(data)))
-- 
2.50.1

